src/ensembl/src/content/app/browser/drawer/drawer-views/transcript-summary/TranscriptSummary.tsx

```diff
@@ -14,70 +14,320 @@
+    | 'stable_id'
+    | 'unversioned_stable_id'
+    | 'symbol'
+    | 'so_term'
+    | 'external_references'
+    | 'slice'
+    | 'spliced_exons'
+    | 'product_generating_contexts'
+  >
+>;
 
-const TranscriptSummary = () => {
-  const ensObject = useSelector(getBrowserActiveEnsObject) as EnsObjectGene;
+type Gene = Required<
+  Pick<
+    GeneFromGraphql,
+    'stable_id' | 'unversioned_stable_id' | 'symbol' | 'name'
+  >
+>;
 
-  if (!ensObject.track) {
-    return null;
+const GENE_AND_TRANSCRIPT_QUERY = gql`
+  query Gene($genomeId: String!, $geneId: String!, $transcriptId: String!) {
+    gene(byId: { genome_id: $genomeId, stable_id: $geneId }) {
+      name
+      stable_id
+      unversioned_stable_id
+      symbol
+    }
+    transcript(byId: { genome_id: $genomeId, stable_id: $transcriptId }) {
+      stable_id
+      unversioned_stable_id
+      so_term
+      symbol
+      external_references {
+        accession_id
+        url
+        source {
+          id
+        }
+      }
+      spliced_exons {
+        relative_location {
+          start
+          end
+        }
+        exon {
+          stable_id
+          slice {
+            location {
+              length
+            }
+          }
+        }
+      }
+      product_generating_contexts {
+        product_type
+        default
+        cds {
+          protein_length
+        }
+        phased_exons {
+          start_phase
+          end_phase
+          exon {
+            stable_id
+          }
+        }
+        product {
+          stable_id
+          external_references {
+            accession_id
+            url
+            source {
+              id
+            }
+          }
+        }
+      }
+      slice {
+        strand {
+          code
+        }
+        location {
+          length
+        }
+      }
+    }
   }
+`;
 
-  // FIXME: this is a temporary function; need to come up with something more robust
-  const getTranscriptTrack = () => {
-    const childTracks = get(ensObject, 'track.child_tracks', []);
+const TranscriptSummary = () => {
+  const ensObjectGene = useSelector(getBrowserActiveEnsObject) as EnsObjectGene;
 
-    return find(childTracks, { track_id: TRANSCRIPT_GENE_NAME }) || null;
-  };
+  const transcriptTrack = ensObjectGene?.track?.child_tracks?.[0];
 
-  // FIXME: this is a very horrible temporary function;
-  // it will break when multiple transcripts are added
-  // but by that time we'll have to do a major refactor for ensObject and ensObject tracks anyway
-  const getTranscriptStableId = () => {
-    return get(ensObject, 'track.child_tracks.0.label', '');
-  };
+  const [shouldShowDownload, showDownload] = useState(false);
 
-  const transcriptTrack = getTranscriptTrack();
+  const { data, loading } = useQuery<{
+    gene: Gene;
+    transcript: Transcript;
+  }>(GENE_AND_TRANSCRIPT_QUERY, {
+    variables: {
+      geneId: ensObjectGene.stable_id,
+      transcriptId: transcriptTrack?.stable_id,
```

Comment:
This isn't working for wheat, because the data from `/api/object/track_list` isn't consistent with Thoas. Ideally, this part of the track panel should be getting its data from Thoas as well:

![image](https://user-images.githubusercontent.com/6834224/107388800-1ad72480-6aee-11eb-9867-8e5d068148de.png)

Since we seem to be in a hurry, we might want to postpone that for a separate PR.

Reply:
Agreed

Reply:
Can you create a ticket, for BE initially

Reply:
@ens-st3 what will the backend need to do? Don't we already get everything we need from Thoas?

Reply:
Was wondering what we could do to get it working for wheat, both short term and longer term. Feels like a JIRA discussion ?